src/main/java/com/notably/model/viewstate/ViewStateModelImpl.java

```diff
@@ -12,10 +17,13 @@
 public class ViewStateModelImpl implements ViewStateModel {
 
     private StringProperty input;
+    private Property<Optional<String>> responseTextProperty;
+
```

Comment:
Nit: Let's remove this newline? In addition, you can actually make all the properties above, `input` and `responseTextProperty` final (since their reference will never change). One last thing: you can also rename `responseTextProperty` into `responseText` only for consistency with the other properties here.

Reply:
ok!